File: Networking/src/main/java/rpc/Request.java
package rpc;

import java.io.Serializable;

public class Request implements Serializable {
    private RequestType type;
    private Object data;

    public RequestType type() { return type; }
    public Object data() { return data; }
    private void type(RequestType type) { this.type = type; }
    private void data(Object data) { this.data = data; }

    @Override
    public String toString() {
        return "Request{" +
                "type=" + type +
                ", data=" + data +
                '}';
    }

    public static class Builder {
        private final Request request = new Request();

        public Builder type(RequestType type) {
            request.type(type);
            return this;
        }

        public Builder data(Object data) {
            request.data(data);
            return this;
        }

        public Request build() { return request; }
    }
}

File: Networking/rpc/Request.cs
using System;

namespace Networking.rpc
{
    [Serializable]
    public class Request
    {
        public RequestType Type { get; set; }
        public object Data { get; set; }
        public override string ToString()
        {
            return "Request{type=" + Type + ", data=" + Data + "}";
        }

        public class Builder
        {
            private readonly Request _request = new();

            public Builder Type(RequestType type)
            {
                _request.Type = type;
                return this;
            }

            public Builder Data(object data)
            {
                _request.Data = data;
                return this;
            }

            public Request Build()
            {
                return _request;
            }
        }
    }
}
